### ISSUE TITLE
Failed to load Reddit2 Dataset

### ISSUE THREAD

**Issue by LemonAndRabbit** [pyg-team/pytorch_geometric#4170]
### 🐛 Describe the bug

I met a problem when loadding Reddit2 Dataset:

When changing `Reddit` in examples/Reddit.py into `Reddit2`

```python
import copy
import os.path as osp

import torch
import torch.nn.functional as F
from tqdm import tqdm

from torch_geometric.datasets import Reddit2
from torch_geometric.loader import NeighborLoader
from torch_geometric.nn import SAGEConv

device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')

path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'Reddit2')
dataset = Reddit2(path)
```

I got the following error:
```text
Traceback (most recent call last):
  File "reddit.py", line 15, in <module>    dataset = Reddit2(path)  File "/home/zy50/anaconda3/envs/pyg/lib/python3.8/site-packages/torch_geometric/datasets/reddit2.py", line 41, in __init__
    super().__init__(root, transform, pre_transform)  File "/home/zy50/anaconda3/envs/pyg/lib/python3.8/site-packages/torch_geometric/data/in_memory_dataset.py", line 57, in __init__    super().__init__(root, transform, pre_transform, pre_filter)
  File "/home/zy50/anaconda3/envs/pyg/lib/python3.8/site-packages/torch_geometric/data/dataset.py", line 88, in __init__    self._process()
  File "/home/zy50/anaconda3/envs/pyg/lib/python3.8/site-packages/torch_geometric/data/dataset.py", line 171, in _process
    self.process()
  File "/home/zy50/anaconda3/envs/pyg/lib/python3.8/site-packages/torch_geometric/datasets/reddit2.py", line 68, in process
    f = np.load(osp.join(self.raw_dir, 'adj_full.npz'))
  File "/home/zy50/anaconda3/envs/pyg/lib/python3.8/site-packages/numpy/lib/npyio.py", line 445, in load
    raise ValueError("Cannot load file containing pickled data "
ValueError: Cannot load file containing pickled data when allow_pickle=False
```

Adding allow_pickle=true in `/home/zy50/anaconda3/envs/pyg/lib/python3.8/site-packages/torch_geometric/datasets/reddit2.py, line 68` doesn't work either:
```text
Processing...
Traceback (most recent call last):
  File "/home/zy50/anaconda3/envs/pyg/lib/python3.8/site-packages/numpy/lib/npyio.py", line 448, in load
    return pickle.load(fid, **pickle_kwargs)
_pickle.UnpicklingError: invalid load key, '<'.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "reddit.py", line 15, in <module>
    dataset = Reddit2(path)
  File "/home/zy50/anaconda3/envs/pyg/lib/python3.8/site-packages/torch_geometric/datasets/reddit2.py", line 41, in __init__
    super().__init__(root, transform, pre_transform)
  File "/home/zy50/anaconda3/envs/pyg/lib/python3.8/site-packages/torch_geometric/data/in_memory_dataset.py", line 57, in __init__
    super().__init__(root, transform, pre_transform, pre_filter)
  File "/home/zy50/anaconda3/envs/pyg/lib/python3.8/site-packages/torch_geometric/data/dataset.py", line 88, in __init__
    self._process()
  File "/home/zy50/anaconda3/envs/pyg/lib/python3.8/site-packages/torch_geometric/data/dataset.py", line 171, in _process
    self.process()
  File "/home/zy50/anaconda3/envs/pyg/lib/python3.8/site-packages/torch_geometric/datasets/reddit2.py", line 68, in process
    f = np.load(osp.join(self.raw_dir, 'adj_full.npz'), allow_pickle=True)
  File "/home/zy50/anaconda3/envs/pyg/lib/python3.8/site-packages/numpy/lib/npyio.py", line 450, in load
    raise IOError(
OSError: Failed to interpret file '/home/zy50/pytorch_geometric/data/Reddit2/raw/adj_full.npz' as a pickle
```

I wonder whether Reddit2 files are broken or I made anything worng?

Thanks a lot for any help.

### Environment

* PyG version: 2.0.3
* PyTorch version: 1.10.2 
* OS: Linux
* Python version: 3.8.12
* How you installed PyTorch and PyG (`conda`, `pip`, source): conda create -n pyg -f pytorch_geometric/conda/pytorch_geometric/pyg.yml

**Comment by rusty1s** [comment]
Thanks for reporting. I think this is due to a change in Google Drive policy that `GoogleDriveDownloader` can no longer handle. I think I fixed it in #4172.

**Comment by LemonAndRabbit** [comment]
Thank you! it works now.
